internal/engine/handler.go

```diff
@@ -57,13 +61,23 @@ func NewExecutorEventHandler(
 	handlerMiddleware []message.HandlerMiddleware,
 	executor Executor,
 ) *ExecutorEventHandler {
-	return &ExecutorEventHandler{
+	eh := &ExecutorEventHandler{
 		evt:                    evt,
 		wgEntityEventExecution: &sync.WaitGroup{},
-		terminationcontext:     ctx,
 		handlerMiddleware:      handlerMiddleware,
 		executor:               executor,
 	}
+	go func() {
+		<-ctx.Done()
+		eh.lock.Lock()
+		defer eh.lock.Unlock()
+
+		for _, cancel := range eh.cancels {
+			(*cancel)()
+		}
+	}()
```

Comment:
If I got this right, you're still waiting for `Done()` on what was originally the `terminationcontext`, but keep track of all downstream contexts to cancel them as well, right?

If yes, was this source of some sort of bug?